### PR TITLE
exclude test for systems without longdouble support

### PR DESCRIPTION
**Pull request by ben-schwen**
Found this test failing while developing with WSL1

No news since test only change

### CODE REVIEW

File: inst/tests/tests.Rraw [hunk 1]
```diff
@@ -17878,7 +17878,9 @@ if (test_bit64) {
   test(2226.1, base::prod(2147483647L,2L), 4294967294)  # just to illustrate that base returns double
   DT = data.table(x=c(lim.integer64(), 2, 1, NA, NA, -2, 4), g=INT(1,2,1,2,1,2,3,3))
   test(2226.2, DT[, prod(x), g],            data.table(g=1:3, V1=as.integer64(c(NA,NA,-8L))))
-  test(2226.3, DT[, prod(x,na.rm=TRUE), g], data.table(g=1:3, V1=as.integer64(c(NA,"9223372036854775807",-8L))))
+  if (test_longdouble) {
```
Comment (MichaelChirico) [inline inst/tests/tests.Rraw:17881]: what happens on `!longdouble` systems?

I saw a segfault in the `r-cmd-check-occasional` logs yesterday, not sure if it's related:

https://github.com/Rdatatable/data.table/actions/runs/9140204969/job/25133296565

Probably best to leave the reasoning here in a comment.

Comment (ben-schwen) [reply inst/tests/tests.Rraw:17881]: On my `!longdouble` with 53 long doubles the values of `lim.integer64()` get rounded to -(2^63) resp. 2^63 and then converted to `NA` here:

https://github.com/Rdatatable/data.table/blob/4045e8bafe2d569de37ac22a71bc0774f6c82526/src/gsumm.c#L1167

File: src/gsumm.c [hunk 2]
```diff
@@ -1139,8 +1139,10 @@ SEXP gprod(SEXP x, SEXP narmArg) {
         const int64_t elem = nosubset ? xd[i] : (irows[i]==NA_INTEGER ? NA_INTEGER64 : xd[irows[i]-1]);
         if (elem==NA_INTEGER64) {
           if (!narm) s[thisgrp] = NA_REAL;
+          Rprintf("CONT: %d %Lf\n", i, s[thisgrp]);
```
Comment (MichaelChirico) [inline src/gsumm.c:1142]: looks like this was included unintentionally

File: inst/tests/tests.Rraw (hunk 1 above)
Comment (MichaelChirico) [reply inst/tests/tests.Rraw:17881]: thanks!

**Review by MichaelChirico — APPROVED** [review]
ready to go after reverting incidental change

**Review by MichaelChirico — APPROVED** [review]
thanks!

**Pull request merged**